hal/modem_api.h

```diff
@@ -0,0 +1,88 @@
+/* mbed Microcontroller Library
```

Comment:
@sg- Kevin explained in detail above. 
Summary:
According to mbed-os guidelines is there is a device available on-chip, you mark it in _device_has:_ block. So you have modem_api only in case if the target _has_ a _DEVICE_MODEM_